includes/class-wcpdf-settings-debug.php

```diff
@@ -161,136 +161,153 @@ public function init_settings() {
+					                 __( 'You can also add <code>&output=html</code> to the URL to apply this on a per-order basis.', 'woocommerce-pdf-invoices-packing-slips' ),
 				)
 			),
 			array(
-				'type'			=> 'setting',
-				'id'			=> 'use_html5_parser',
-				'title'			=> __( 'Use alternative HTML5 parser to parse HTML', 'woocommerce-pdf-invoices-packing-slips' ),
-				'callback'		=> 'checkbox',
-				'section'		=> 'debug_settings',
-				'args'			=> array(
-					'option_name'	=> $option_name,
-					'id'			=> 'use_html5_parser',
+				'type'     => 'setting',
+				'id'       => 'use_html5_parser',
+				'title'    => __( 'Use alternative HTML5 parser to parse HTML', 'woocommerce-pdf-invoices-packing-slips' ),
+				'callback' => 'checkbox',
+				'section'  => 'debug_settings',
+				'args'     => array(
+					'option_name' => $option_name,
+					'id'          => 'use_html5_parser',
 				)
 			),
 			array(
-				'type'			=> 'setting',
-				'id'			=> 'log_to_order_notes',
-				'title'			=> __( 'Log to order notes', 'woocommerce-pdf-invoices-packing-slips' ),
-				'callback'		=> 'checkbox',
-				'section'		=> 'debug_settings',
-				'args'			=> array(
-					'option_name'	=> $option_name,
-					'id'			=> 'log_to_order_notes',
-					'description'	=> __( 'Log PDF document creation to order notes.', 'woocommerce-pdf-invoices-packing-slips' ),
+				'type'     => 'setting',
+				'id'       => 'log_to_order_notes',
+				'title'    => __( 'Log to order notes', 'woocommerce-pdf-invoices-packing-slips' ),
+				'callback' => 'checkbox',
+				'section'  => 'debug_settings',
+				'args'     => array(
+					'option_name' => $option_name,
+					'id'          => 'log_to_order_notes',
+					'description' => __( 'Log PDF document creation to order notes.', 'woocommerce-pdf-invoices-packing-slips' ),
 				)
 			),
 			array(
-				'type'			=> 'setting',
-				'id'			=> 'disable_preview',
-				'title'			=> __( 'Disable document preview', 'woocommerce-pdf-invoices-packing-slips' ),
-				'callback'		=> 'checkbox',
-				'section'		=> 'debug_settings',
-				'args'			=> array(
-					'option_name'	=> $option_name,
-					'id'			=> 'disable_preview',
-					'description'	=> __( 'Disables the document preview on the plugin settings pages.', 'woocommerce-pdf-invoices-packing-slips' ),
+				'type'     => 'setting',
+				'id'       => 'pretty_document_links',
+				'title'    => __( 'Pretty document links', 'woocommerce-pdf-invoices-packing-slips' ),
+				'callback' => 'checkbox',
+				'section'  => 'debug_settings',
+				'args'     => array(
+					'option_name' => $option_name,
+					'id'          => 'pretty_document_links',
+					'description' => sprintf(
+						/* translators: <a> tags */
+						__( 'Changes the document links to a prettier URL scheme. (Saving the %1$spermalinks%2$s might be required)', 'woocommerce-pdf-invoices-packing-slips' ),
```

Comment:
This should be done automatic too, when the debug settings are saved! (`flush_rewrite_rules()`  should do the trick?)

Reply:
Doesn't work, and because of that I found this: https://wordpress.stackexchange.com/a/47750

Let me know your thoughts.

Reply:
Have you tried that suggestion (to do it on the next `init`, either with a transient or with AS)?

Reply:
The fact that it's expensive doesn't matter if you only run it when saving the settings.

Reply:
Agreed